core/amber/src/main/python/core/architecture/packaging/input_manager.py

```diff
@@ -97,17 +108,30 @@ def process_data_payload(
             yield SenderChange(current_channel_id)
 
         if isinstance(payload, DataFrame):
-            for field_accessor in ArrowTableTupleProvider(payload.frame):
-                yield Tuple(
-                    {name: field_accessor for name in payload.frame.column_names},
-                    schema=self._ports[
-                        self._channels[self._current_channel_id].port_id
-                    ].get_schema(),
-                )
+            yield from self._process_data(payload.frame)
+        elif isinstance(payload, MarkerFrame):
+            yield from self._process_marker(payload.frame)
+        else:
+            raise NotImplementedError()
 
-        elif isinstance(payload, MarkerFrame) and isinstance(
-            payload.frame, EndOfUpstream
-        ):
+    def _process_data(self, table: Table) -> Iterator[Tuple]:
+        schema = self._ports[
+            self._channels[self._current_channel_id].port_id
+        ].get_schema()
+        for field_accessor in ArrowTableTupleProvider(table):
+            yield Tuple(
+                {name: field_accessor for name in table.column_names}, schema=schema
+            )
+
+    def _process_marker(self, marker: Marker) -> Iterator[InternalMarker]:
+        if isinstance(marker, State):
+            yield marker
+        if isinstance(marker, StartOfUpstream):  # StartOfInputChannel()
+            if not self.started:
+                yield StartOfOutputPorts()
+            self.started = True
+            yield StartOfUpstream()  # StartOfInputChannel()
```

Comment:
for those marker names with comments, do you want to change the name in this PR? it is not good to have `upstream` in the name.

Reply:
Renamed.